src/SELF_MappedScalar_3D_t.f90

```diff
@@ -123,308 +123,306 @@ subroutine SetInteriorFromEquation_MappedScalar3D_t(this,geometry,time)
 
   endsubroutine SetInteriorFromEquation_MappedScalar3D_t
 
-  ! subroutine MPIExchangeAsync_MappedScalar3D_t(this,decomp,mesh,resetCount)
-  !   implicit none
-  !   class(MappedScalar3D_t),intent(inout) :: this
-  !   type(MPILayer),intent(inout) :: decomp
-  !   type(Mesh3D),intent(in) :: mesh
-  !   logical,intent(in) :: resetCount
-  !   ! Local
-  !   integer :: e1,s1,e2,s2,ivar
-  !   integer :: globalSideId,r2
-  !   integer :: iError
-  !   integer :: msgCount
-
-  !   if(decomp%mpiEnabled) then
-  !     if(resetCount) then
-  !       msgCount = 0
-  !     else
-  !       msgCount = decomp%msgCount
-  !     endif
-
-  !     do ivar = 1,this%nvar
-  !       do e1 = 1,this%nElem
-  !         do s1 = 1,6
-
-  !           e2 = mesh%sideInfo(3,s1,e1) ! Neighbor Element
-  !           if(e2 > 0) then
-  !             r2 = decomp%elemToRank(e2) ! Neighbor Rank
-
-  !             if(r2 /= decomp%rankId) then
-
-  !               ! to do : create unique tag for each side and each variable
-  !               ! tag = globalsideid + nglobalsides*ivar
-  !               s2 = mesh%sideInfo(4,s1,e1)/10
-  !               globalSideId = abs(mesh%sideInfo(2,s1,e1))
-
-  !               msgCount = msgCount+1
-  !               call MPI_IRECV(this%extBoundary(:,:,s1,e1,ivar), &
-  !                              (this%interp%N+1)*(this%interp%N+1), &
-  !                              decomp%mpiPrec, &
-  !                              r2,globalSideId, &
-  !                              decomp%mpiComm, &
-  !                              decomp%requests(msgCount),iError)
-
-  !               msgCount = msgCount+1
-  !               call MPI_ISEND(this%boundary(:,:,s1,e1,ivar), &
-  !                              (this%interp%N+1)*(this%interp%N+1), &
-  !                              decomp%mpiPrec, &
-  !                              r2,globalSideId, &
-  !                              decomp%mpiComm, &
-  !                              decomp%requests(msgCount),iError)
-  !             endif
-  !           endif
+  subroutine MPIExchangeAsync_MappedScalar3D_t(this,mesh,resetCount)
+    implicit none
+    class(MappedScalar3D_t),intent(inout) :: this
+    type(Mesh3D),intent(inout) :: mesh
+    logical,intent(in) :: resetCount
+    ! Local
+    integer :: e1,s1,e2,s2,ivar
+    integer :: globalSideId,r2,tag
+    integer :: iError
+    integer :: msgCount
+
+    if(resetCount) then
```

Comment:
Same as above.. remove the resetcount option.

Reply:
Resolved in [f24f99d](https://github.com/FluidNumerics/SELF/pull/53/commits/f24f99db9e33d33c61cba24a30e9e8176c9a85ea)